### PR TITLE
Sg/stack master v2 urp coat [skip ci]

### PR DESCRIPTION
**Pull request by eh-unity**
**Guide** : https://github.com/Unity-Technologies/Graphics/blob/pr-template-correction/.github/pr-read.png.md

**Display Addon** : https://userstyles.org/styles/182991/unity-graphics-pr-readme

# Purpose of this PR

> Why is this PR needed, what hard problem is it solving/fixing?

Integrate URP clear coat with refactored shadergraph master-stack system.

base is set to sg/stack-master-v2-upgrade-hdrp in order to see the actual change.

The plan:
Needs stack-mast 'sg/stack-master-v2-upgrade-hdrp' and urp clear coat ' graphics/urp-clear-coat' to land first.
Merge master into this.
Then fix pbrforwardpass.hlsl to use actual api coat in and lighting.hlsl instead of the placehoder api in this one.
Test.
Merge into mater.

# Testing status

## Links
**Yamato**: (Select your branch) https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics
> Any test projects or documents to go with this to help reviewers?

# Comments to reviewers
> Notes for the reviewers you have assigned.

### CODE REVIEW

File: com.unity.shadergraph/Editor/Generation/TargetResources/Fields.cs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-﻿namespace UnityEditor.ShaderGraph
+espace UnityEditor.ShaderGraph
```
Comment (Kink3d) [inline com.unity.shadergraph/Editor/Generation/TargetResources/Fields.cs:1]: Typo

File: com.unity.shadergraph/Editor/Generation/TargetResources/BlockFields.cs [hunk 2]
```diff
@@ -1,56 +1,60 @@
-﻿using UnityEngine;
```
Comment (Kink3d) [inline com.unity.shadergraph/Editor/Generation/TargetResources/BlockFields.cs:1]: Can probably revert this change

Comment (eh-unity) [reply com.unity.shadergraph/Editor/Generation/TargetResources/BlockFields.cs:1]: Yeah, I don't even know how this happened or what's changed.
I think it's unity-meta and .editorconfig stuff.

What I did is just add CoatMask and CoatSmoothness BlockFields.

Comment (eh-unity) [reply com.unity.shadergraph/Editor/Generation/TargetResources/BlockFields.cs:1]: I think line-ends changed.

File: com.unity.shadergraph/Editor/Generation/TargetResources/Fields.cs (hunk 1 above)
Comment (eh-unity) [reply com.unity.shadergraph/Editor/Generation/TargetResources/Fields.cs]: fixed

File: com.unity.shadergraph/Editor/Generation/TargetResources/BlockFields.cs (hunk 2 above)
Comment (eh-unity) [reply com.unity.shadergraph/Editor/Generation/TargetResources/BlockFields.cs:1]: Yeah, I don't even know how this happened or what's changed, looks like line-ends.
I think it's .editorconfig stuff I've got set up.

What I did is just add CoatMask and CoatSmoothness BlockFields.

**Comment by phi-lira** [comment]
Closing, next iteration https://github.com/Unity-Technologies/Graphics/pull/1295/
